### ISSUE TITLE
Review and implement Lely habitat classification

### ISSUE THREAD

**Issue by mikkohei13** [luomus/laji#572]
Blocks https://github.com/luomus/meta/issues/177 and https://github.com/luomus/mobile-vihko-app/issues/63

This is about field https://schema.laji.fi/property/MY.habitat

**NOTE:** This is the latest document, do any changes here and nowhere else: https://docs.google.com/spreadsheets/d/1zQ6R-gkJVly8zcgpOLGHHNg0DI2lPktF/edit?usp=sharing&ouid=114875904399264808995&rtpof=true&sd=true

- [x] Review classification, which is at 
   - Discussion at https://luomus-ict.slack.com/archives/C01Q5LZ2DMZ/p1728026426125609
- [x] Review classification again, once Syke worked on it
- [x] Finalize the classification
   - https://github.com/luomus/meta/blob/main/files/habitat_classification_v1.0.tsv
- [x] Finalize the classification for realz
   - [x] Get missing translations. Here: https://helsinkifi-my.sharepoint.com/:x:/g/personal/mqheikki_ad_helsinki_fi/EVCcGl2d35hKtWy_isWxIIEBlsI5jZJt15urawF--0CsvA?e=qgeEvE
   - [x] Get translations that are still missing
   - [x] Add missing translations to schema
- [x] Add missing enum values to schema
- [ ] Combine MY.habitatEnumValue87 (Karut tai keskiravinteinen kalliot ("tavalliset kalliot")) to MY.habitatEnumValue146 Keskiravinteinen kallio:
   - [ ] Migrate data on Vihko.
   - [ ] Delete MY.habitatEnumValue87 from schema.

**Comment by mikkohei13** [comment]
Blocked by Syke reviewing the classification.

**Comment by mikkohei13** [comment]
New version of the classification (2024-10-24): https://helsinkifi-my.sharepoint.com/:x:/g/personal/mqheikki_ad_helsinki_fi/EQWuspQQsP9Cld8WGPOLT60BGqQfoz4dz8REPR7GUSVUWg?e=dS9k4H

Note: this has some level-1 classes that have only one level-2 subclass, e.g. Tervaleppämetsä -> Tervaleppämetsä. In this case, we can link level1 to level-3 directly, so users need fewer clicks.

Need to double-check relations, e.g. Jalopuumetsä, Jalopuulehto & Jalopuustoinen kangasmetsä feel confusing.

Names are harmonized to singular.

**Comment by mikkohei13** [comment]
Can be developed to Vihko & Mobiilivihko using https://github.com/luomus/meta/blob/main/files/habitat_classification_v1.0.tsv

**Comment by mikkohei13** [comment]
Todo in 2025: additional attributes.

**Comment by mikkohei13** [comment]
Moved additional attributes to another task https://github.com/luomus/meta/issues/199

**Comment by Tea-shroom** [comment]
[like]  von Bonsdorff-Salminen, Tea K reacted to your message:
________________________________

**Comment by mikkohei13** [comment]
Latest version of the classification: https://helsinkifi-my.sharepoint.com/:x:/r/personal/mqheikki_ad_helsinki_fi/_layouts/15/Doc.aspx?sourcedoc=%7BB412CE0D-ADC8-4A45-94F3-9B274BCCCEC4%7D&file=Habitaatit-20250205.xlsx&action=default&mobileredirect=true

**Comment by mikkohei13** [comment]
Next someone should add new values and translations to schema.

**Comment by olzraiti** [comment]
Add new values and translations to schema.

There are a lot of missing translations though. I marked them with bold here: https://helsinkifi-my.sharepoint.com/:x:/g/personal/olzraiti_ad_helsinki_fi/ESH9ELRED1pDvAJ30WkyaFgByr3AgkMyP-PyXGaJ8vIUvQ?e=pg2Nru

**Comment by mikkohei13** [comment]
Only translations to schema to be done?

**Comment by olzraiti** [comment]
The translations to the schema are done according to the provided spreadsheet. But there are still translations missing in the spreadsheet, so if somebody provides the missing translations then that's to be done still.

**Comment by mikkohei13** [comment]
NOTE: Use this link to see latest habitats: https://docs.google.com/spreadsheets/d/1zQ6R-gkJVly8zcgpOLGHHNg0DI2lPktF/edit?usp=sharing&ouid=114875904399264808995&rtpof=true&sd=true

**Comment by Tea-shroom** [comment]
[like]  von Bonsdorff-Salminen, Tea K reacted to your message:
________________________________

**Comment by mikkohei13** [comment]
Todo:

- [ ] Combine
MY.habitatEnumValue55 (Perinneympäristöt tai muut ihmisen muuttamat ympäristöt)
to
MY.habitatEnumValue79 (Uuselinympäristöt)

- [ ] Combine
MY.habitatEnumValue63 (Kallioketo)
to
MY.habitatEnumValue43 (Kallioketo)

- [ ] Combine
MY.habitatEnumValue87 (Karut tai keskiravinteinen kalliot ("tavalliset kalliot"))
to
(to be decided)

- [ ] Check English and Swedish translations of all values.

**Comment by olzraiti** [comment]
Done

**Comment by olzraiti** [comment]
Received an API error email about an user failing to update a named place (`MNP.32935`), because it has `MY.habitatEnumValue55`. Named places weren't migrated maybe? It's probably a pretty common that we forget to do migrations for named places also.
